apps/wpcom-block-editor/src/calypso/iframe-bridge-server.js

```diff
@@ -17,6 +17,63 @@ import tinymce from 'tinymce/tinymce';
  */
 import { inIframe, sendMessage } from './utils';
 
+/**
+ * Monitors Gutenburg for when an editor is opened with content originally authored in the classic editor.
+ *
+ * @param {MessagePort} calypsoPort Port used for communication with parent frame.
+ */
+function triggerConversionPrompt( calypsoPort ) {
+	function addConversionListener( classicBlock ) {
+		// Listen for user response
+		calypsoPort.addEventListener( 'message', onConversionResponse, false );
+		calypsoPort.start();
+
+		function onConversionResponse( message ) {
+			const action = get( message, 'data.action' );
+			const payload = get( message, 'data.payload' );
+
+			if ( action !== 'conversionResponse' ) {
+				return;
+			}
+
+			// Removing the listener regardless of which action was taken
+			calypsoPort.removeEventListener( 'message', onConversionResponse, false );
+
+			// Converting the block if the user selected the "Convert" button in the prompt
+			if ( payload.confirmed === true ) {
+				const unsubscribe = subscribe( () => {
```

Comment:
Since we not selecting anything from the state, I don't think we need this subscriber:

```js
if ( payload.confirmed ) {
	dispatch( 'core/editor' ).replaceBlock(
		classicBlock.clientId,
		rawHandler( {
			HTML: classicBlock.originalContent,
		} )
	);
}
```